### PR TITLE
 [FIX] mail: link-preview on public channel invitations

### PR DESCRIPTION
**Pull request by jayp-odoo**
**Current behavior before PR:**

When link-preview is done on public channel invitation links it posts a guest
joined the channel​ message in the public channel.

**Desired behavior after PR is merged:**

now we are ignoring public channel invitations links for link-preview, so it
does not post guest joined the channel

backported PR: https://github.com/odoo/odoo/pull/178932 which handle's ignore link-preview.

task-4083161

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by alexkuhn — COMMENTED** [review]
Would clarify in commit message that this is a backport of some 18.0 code.

Also the diff makes use of newer URLs in 18.0. This is a 17.0 PR so it should not test the new URLs nor handle them.

**Comment by jayp-odoo** [comment]
@alexkuhn i have updated PR can you please look into it. thanks :+1:

**Review by alexkuhn — COMMENTED** [review]
Having the link preview of publicly available chat links is useful: it could show the chat name in preview.

The "joined" notification is an undesirable side-effect, but I think we can prevent the joined notification while still show the link preview.

For example, we could call the `get_link_preview_from_url` with a special param, like `as_link_preview=true`, which is subtlety managed by public link in a way to assume this isn't an actual GET from a genuine user, thus it forces showing of welcome page without joining the conversation.

Doing so would allow to support link preview of public Discuss welcome page, which has value as a public URL.

**Comment by alexkuhn** [comment]
This is a backport so as a FIX this is good, in master we could improve it

**Review by alexkuhn — COMMENTED** [review]
LGTM otherwise

File: addons/mail/tests/test_link_preview.py [hunk 1]
```diff
@@ -154,3 +154,29 @@ def test_link_preview_no_content_type(self):
...
+    def test_link_preview_ignore_internal_link(self):
+        """Test internal links are properly ignored from link preview."""
+        with patch.object(requests.Session, "get", self._patch_with_og_properties), patch.object(
+            requests.Session, "head", self._patch_head_html
+        ):
+            urls = [
+                ("http://localhost:8069/", "http://localhost:8069/chat/5/bFtIfYHRco", 0),
+                ("https://www.odoo.com/", "https://www.odoo.com/chat/", 0),
+                ("https://www.odoo.com/", "https://www.odoo.com/chat/5/bFtIfYHRco", 0),
+                ("https://www.odoo.com/", "https://www.odoo.com/odoo/", 1),
```
[the remark this answers is not in the document]
Comment (jayp-odoo) [reply addons/mail/tests/test_link_preview.py]: Yes, I have updated the code can you please look into it. thanks :)

**Review by alexkuhn — APPROVED** [review]
@robodoo r+ override=ci/security
`markup()` to craft a message post with a URL in test, in order to generate a link preview

**Pull request closed without merging**